src/sage/topology/moment_angle_complex.py

```diff
@@ -724,8 +721,8 @@ def euler_characteristic(self):
             sage: Z.euler_characteristic()
             1
         """
-        betti_numbers = self.betti()
-        return ZZ.sum((-1)**n * betti_numbers[n] for n in range(self.dimension() + 1))
+        sc = self.simplicial_complex()
+        return 1 if sc.dimension() + 1 == len(sc.vertices()) else 0
```

Comment:
```suggestion
        return ZZ.one() if sc.dimension() + 1 == len(sc.vertices()) else ZZ.zero()
```
We shouldn't return `int`s.